canary/generator/download/query.py

```diff
@@ -0,0 +1,13 @@
+def build_query_string(dates, metric, channel, version, os=None, os_version=None,
+                       application=None, architecture=None):
+    query_base = 'https://aggregates.telemetry.mozilla.org/aggregates_by/build_id/channels/'
+    date_str = '%2C'.join(dates)
+    os = '&os=' + os if os else ''
+    os_version = '&osVersion=' + os_version if os_version else ''
+    application = '&application=' + application if application else ''
+    architecture = '&architecture=' + architecture if architecture else ''
```

Comment:
Each of these lines uses the same variable 3 times.

How about something like this instead:
```python
query = query_base + ...
if os:
  query += '&os=' + os
...
```